### PR TITLE
Handled exception allows user-defined custom properties

### PR DESCRIPTION
**Pull request by dhei**

### CODE REVIEW

File: sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java [hunk 1]
```diff
@@ -206,13 +206,28 @@ static synchronized void unsetInstance() {
...
     /**
      * Track an exception.
-     * TODO the backend does not support that service yet, will be public method later.
      *
      * @param throwable An exception.
      */
     @SuppressWarnings("SameParameterValue")
-    static void trackException(@NonNull Throwable throwable) {
-        getInstance().queueException(throwable);
+    public static void trackException(@NonNull Throwable throwable) {
```
Comment (guperrot) [inline sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java]: oh no don't make it public yet, the public visibility is only for Xamarin (And wrapper sdk exception manager), restore package private and the TODO

File: sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java [hunk 2]
```diff
@@ -206,13 +206,28 @@ static synchronized void unsetInstance() {
...
+     * The name parameter can not be null or empty. Maximum allowed length = 256.
+     * The properties parameter maximum item count = 5.
+     * The properties keys can not be null or empty, maximum allowed key length = 64.
+     * The properties values can not be null, maximum allowed value length = 64.
+     * Any length of name/keys/values that are longer than each limit will be truncated.
+     *
+     * @param throwable An exception.
+     * @param properties Optional properties.
+     */
+    public static void trackException(@NonNull Throwable throwable, Map<String, String> properties) {
```
Comment (guperrot) [inline sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java]: same comment about visibility

File: sdk/appcenter-crashes/src/test/java/com/microsoft/appcenter/crashes/CrashesTest.java [hunk 3]
```diff
@@ -524,6 +528,60 @@ public boolean matches(Object item) {
                 return item instanceof HandledErrorLog && EXCEPTION.getMessage().equals(((HandledErrorLog) item).getException().getMessage());
             }
         }), eq(crashes.getGroupName()));
+        reset(mockChannel);
+
+        Crashes.trackException(EXCEPTION, new HashMap<String, String>() {{
```
Comment (guperrot) [inline sdk/appcenter-crashes/src/test/java/com/microsoft/appcenter/crashes/CrashesTest.java]: always start a new block of code with a comment

File: sdk/appcenter-crashes/src/test/java/com/microsoft/appcenter/crashes/CrashesTest.java [hunk 4]
```diff
@@ -625,51 +618,44 @@ public void trackExceptionForWrapperSdk() {
...
                         && ((HandledErrorLog) item).getProperties().size() == 5;
             }
         }), eq(crashes.getGroupName()));
         reset(mockChannel);
-
         final String longerMapItem = generateString(ErrorLogHelper.MAX_PROPERTY_ITEM_LENGTH + 1, '*');
         WrapperSdkExceptionManager.trackException(exception, new HashMap<String, String>() {{
             put(longerMapItem, longerMapItem);
         }});
         verify(mockChannel).enqueue(argThat(new ArgumentMatcher<Log>() {
```
Comment (guperrot) [inline sdk/appcenter-crashes/src/test/java/com/microsoft/appcenter/crashes/CrashesTest.java]: If there is no blank line before @Override, please add one.
We put blank lines before annotations, between each fields and methods, or before any comment.

**Comment by guperrot** [comment]
@dhei looks good, just add blank lines before @Override or other annotations and good to go.

**Comment by guperrot** [comment]
@dhei for commit messages please use upper case for the very first character

File: sdk/appcenter-crashes/src/test/java/com/microsoft/appcenter/crashes/CrashesTest.java (hunk 4 above)
Comment (dhei) [reply sdk/appcenter-crashes/src/test/java/com/microsoft/appcenter/crashes/CrashesTest.java]: thanks, will follow keep in mind of this convention.

**Pull request merged**